// Copyright 2022 Google LLC
//
// Use of this source code is governed by a BSD-style
// license that can be found in the LICENSE file or at
// https://developers.google.com/open-source/licenses/bsd

syntax = "proto3";

package visionai.testing;

message EchoRequest {
  string message = 1;
}

message EchoResponse {
  string message = 1;
}

service EchoService {
  // Returns a response that was present in the request
  rpc Echo(EchoRequest) returns (EchoResponse);
  // A sequence of requests followed by one response.
  rpc ClientStreamingEcho(stream EchoRequest) returns (EchoResponse);
  // One request followed by a sequence of responses.
  rpc ServerStreamingEcho(EchoRequest) returns (stream EchoResponse);
  // A sequence of requests followed by a sequence of responses.
  rpc BidiStreamingEcho(stream EchoRequest) returns (stream EchoResponse);
}

message FooRequest {
  string arg = 1;
}

message FooResponse {
  string arg = 1;
}

service FooService {
  // Used for testing normal invocations.
  rpc Foo(FooRequest) returns (FooResponse);
}
